controllers/iamrole_controller.go

```diff
@@ -140,16 +141,30 @@ func (r *IamroleReconciler) HandleReconcile(ctx context.Context, req ctrl.Reques
 		ns = *ns2
 	}
 
+	// Figure out the name of the IAM Role itself that will be created based on the Iamrole resource name, namespace,
+	// and the configuration settings.
 	roleName, err := utils.GenerateRoleName(ctx, iamRole, *config.Props, &ns)
-	log.V(1).Info("roleName constructed successfully", "roleName", roleName)
-
 	if err != nil {
 		r.Recorder.Event(iamRole, v1.EventTypeWarning, string(iammanagerv1alpha1.Error), "Unable to construct iam role name to error "+err.Error())
 		// It is not clear to me that we want to requeue here - as this is a fairly permanent
 		// error. Is there a better pattern here?
 		return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
+	// Determine whether or not this Iamrole is supposed to be connected to a ServiceAccount or not. This is
```

Comment:
this case of "not managing a ServiceAccount" seems weird - like, why use this if you are not managing a ServiceAccount?

we don't have to deal with it here, but it might be a good idea to try and decouple the ServiceAccount management and the IAMRole management even more - e.g. have the flow be something like:
1. make sure the IAMRole is in the correct state
2. decide if we need to manage a serviceaccount
3. if we do, do whatever serviceaccount managment we need to

Reply:
Agreed. First, to set the stage, the reason they have the "should I manage the SA" flag is because you have to set a specific annotation on the Iamrole resource to tell it to connect the Iamrole to a ServiceAccount. There are many reasons why you might choose to manage an AWS IAMRole and not actually connect it to an SA.

I tried to move the logic for "if I should manage the SA" into the `k8s/rbac.go` file.. but it created a dependency loop. The `k8s/rbac.go` code is inside the `k8s` module. The `k8s` module is loaded up by the `awsapi` module so that it can get some information about how it should connect to AWS. So if you are in the `k8s` module and try to load up the `awsapi` module, you get a dependency loop. :facepalm:

I started down the path of trying to separate these ... but I felt like I was unwinding something I shouldn't touch yet :/

Reply:
yeah any sort of config should probably be being passed into the `k8s` module from here, instead of directly loading it from `awsapi`.

sounds like a battle for another day